packages/artillery/lib/cmds/run.js

```diff
@@ -389,13 +378,31 @@ RunCommand.runCommandImplementation = async function (flags, argv, args) {
       shuttingDown = true;
       global.artillery.globalEvents.emit('shutdown:start', opts);
 
+      // Run beforeExit first, and then onShutdown
+
+      const ps = [];
+      for (const e of global.artillery.extensionEvents) {
+        const testInfo = { endTime: Date.now() };
+        if (e.ext === 'beforeExit') {
+          ps.push(
+            e.method({
+              report: finalReport,
+              flags,
+              runnerOpts,
+              testInfo
+            })
+          );
+        }
+      }
+      await Promise.allSettled(ps);
+
+      const ps2 = [];
```

Comment:
nit: name `ps2` something more meaningful like `onShutdownPromises`

Reply:
My approach is the length of the name should be proportionate to the "reach" of that name. A variable for a loop counter should have a single letter name like `i`. A variable that's only referenced within a screenful of code where it's declared does not need to be long. A name that's going to be referenced from multiple modules/files/functions should probably be descriptive. I'm sure there's a [C2](https://wiki.c2.com/) page on that somewhere but can't dig it up right now!